alibi/explainers/integrated_gradients.py

```diff
@@ -388,84 +468,120 @@ def explain(self,
                            "Not defining the target may lead to incorrect values for the attributions."
                            "Targets can be either the true classes or the classes predicted by the model.")
 
-        nb_samples = len(X)
+        if not isinstance(X, list):
+            X = [X]
+            baselines = [baselines]
 
-        # format and check inputs and targets
-        baselines = _format_input_baseline(X, baselines)
-        target = _format_target(target, nb_samples)
+        if len(X) != len(baselines):  # type: ignore
+            raise ValueError("Length of 'X' must match leght of 'baselines'. "
+                             "Found len(X): {}, len(baselines): {}".format(len(X), len(baselines)))  # type: ignore
+
+        assert max([len(x) for x in X]) == min([len(x) for x in X])
+        nb_samples = len(X[0])
 
         # defining integral method
         step_sizes_func, alphas_func = approximation_parameters(self.method)
         step_sizes, alphas = step_sizes_func(self.n_steps), alphas_func(self.n_steps)
-
-        # construct paths and prepare batches
-        paths = np.concatenate([baselines + alphas[i] * (X - baselines) for i in range(self.n_steps)], axis=0)
+        target = _format_target(target, nb_samples)
         if target is not None:
             target_paths = np.concatenate([target for _ in range(self.n_steps)], axis=0)
```

Comment:
`target_paths` is not defined for regression models which currently fails the tests as it's a required argument for `_calculate_sum_int`